### PR TITLE
Refactoring recommendations

### PR DESCRIPTION
**Pull request by tatitati**

### CODE REVIEW

File: src/RecEng/RecEngService.php [hunk 1]
```diff
@@ -49,15 +43,14 @@ class RecEngService
      */
     private $cache;
 
-    public function __construct(Client $client, string $audioKey, string $videoKey, ProgrammesService $programmesService, UrlGeneratorInterface $router, LoggerInterface $logger, CacheInterface $cache)
```
Comment (tatitati) [inline src/RecEng/RecEngService.php]: the client has been moved to the adapter (which makes the http request)

File: src/RecEng/RecEngService.php [hunk 2]
```diff
@@ -92,35 +85,18 @@ public function getRecommendations(
             return $cacheItem->get();
         }
 
-
-        $recEngKey = $programmeEpisode->isVideo() ? $this->videoKey : $this->audioKey;
+        $recEngKey = $this->recEngKeysManagerValue->getKeyFromProgramme($programmeEpisode->isVideo());
```
Comment (tatitati) [inline src/RecEng/RecEngService.php]: In order to protect the keys we create a object value, which provides the right key based on the programme type.

File: src/RecEng/RecEngService.php [hunk 3]
```diff
@@ -92,35 +85,18 @@ public function getRecommendations(
...
 
         $requestUrl = $this->router->generate(
             'receng',
             array('key' => $recEngKey, 'id' => (string) $programmePid),
             UrlGeneratorInterface::ABSOLUTE_URL
         );
 
-        try {
```
Comment (tatitati) [inline src/RecEng/RecEngService.php]: dependency leaks (guzzle) moved to the adapter which makes the http request

File: src/RecEng/RecEngService.php [hunk 4]
```diff
@@ -92,35 +85,18 @@ public function getRecommendations(
...
-        if ($response->getStatusCode() !== 200) {
-            return [];
-        }
+        /** @var Pid[] $pidsRecommendations */
+        $pidsRecommendations = $this->recEngAdapter->consume($requestUrl, $limit);
 
-        $recEngRespose = $response->getBody()->getContents();
-
-        try {
-            $pids = $this->parseResult($recEngRespose, $limit);
```
Comment (tatitati) [inline src/RecEng/RecEngService.php]: the adapter has a translator to convert/map to our domain concepts (pids)

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/RecEng/RecEngService.php [hunk 5]
```diff
@@ -92,33 +85,15 @@ public function getRecommendations(
...
-        }
-
-        if ($response->getStatusCode() !== 200) {
-            return [];
-        }
-
-        $recEngRespose = $response->getBody()->getContents();
-
-        try {
-            $pids = $this->parseResult($recEngRespose, $limit);
```
Comment (tatitati) [inline src/RecEng/RecEngService.php]: The parsing/mapping process is done now in the translator, which convert external concepts to our domain.

**Comment by BjornTwachtmann** [comment]
I don't think this refactoring is really needed, sorry mate :(

**Pull request closed without merging**